lib/jekyll/configuration.rb

```diff
@@ -98,7 +98,7 @@ def from(user_config)
     #
     # Return a copy of the hash where all its keys are strings
     def stringify_keys
-      reduce({}) { |hsh, (k, v)| hsh.merge(k.to_s => v) }
+      reduce({}) { |hsh, (k, v)| hsh.merge!(k.to_s => v) }
```

Comment:
I'm going to suggest my good ol' friend `each_with_object` here instead:

```ruby
>> conf = {foo: "bar", "baz" => "boz"}
=> {:foo=>"bar", "baz"=>"boz"}
>> conf.each_with_object({}) { |(k,v), hash| hash[k.to_s] = v }
=> {"foo"=>"bar", "baz"=>"boz"}
```

Reply:
Is there a difference in the performance?

Reply:
Even if there is, `each_with_object` is easier to understand.

Reply:
I checked.. and I can confirm that there is a performance issue with `reduce.merge!` as well..
For future readers, the reason is because `k.to_s => v` in `merge!(k.to_s => v)` is a new `Hash` in each iteration..!!